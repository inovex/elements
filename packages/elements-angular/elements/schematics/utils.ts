import { virtualFs, workspaces } from '@angular-devkit/core';
import { SchematicsException, Tree } from '@angular-devkit/schematics';

const CONFIG_PATH = 'angular.json';

function createHost(tree: Tree): workspaces.WorkspaceHost {
  return {
    async readFile(path: string): Promise<string> {
      const data = tree.read(path);
      if (!data) {
        throw new SchematicsException('File not found.');
      }
      return virtualFs.fileBufferToString(data);
    },
    async writeFile(path: string, data: string): Promise<void> {
      return tree.overwrite(path, data);
    },
    async isDirectory(path: string): Promise<boolean> {
      return !tree.exists(path) && tree.getDir(path).subfiles.length > 0;
    },
    async isFile(path: string): Promise<boolean> {
      return tree.exists(path);
    },
  };
}

export function readConfig(host: Tree) {
  const sourceText = host.read(CONFIG_PATH)?.toString('utf-8') || '';
  return JSON.parse(sourceText);
}

export function writeConfig(host: Tree, config: JSON) {
  host.overwrite(CONFIG_PATH, JSON.stringify(config, null, 2));
}

export function getAngularAppConfig(
  config: any,
  projectName: string
): any | never {
  if (!config.projects.hasOwnProperty(projectName)) {
    throw new SchematicsException(`Could not find project: ${projectName}`);
  }

  return config.projects[projectName];
}

export async function getProjectName(tree: Tree) {
  const host = createHost(tree);
  const { workspace } = await workspaces.readWorkspace('/', host);
  return workspace.extensions.defaultProject as string;
}
