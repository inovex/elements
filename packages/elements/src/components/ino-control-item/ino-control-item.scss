@use '../styles/mdc-customize';
@use '@material/list/mdc-list';
@use '@material/form-field/mdc-form-field';
@use '@material/checkbox/mdc-checkbox';
