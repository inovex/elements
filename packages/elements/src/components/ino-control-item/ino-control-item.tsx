import {
  Component,
  ComponentInterface,
  Element,
  Event,
  EventEmitter,
  Host,
  Prop,
  h,
} from '@stencil/core';

/**
 * @slot default - Any element
 */
@Component({
  tag: 'ino-control-item',
  styleUrl: 'ino-control-item.scss',
  shadow: false,
})
export class InoControlItem implements ComponentInterface {
  private inputEl: HTMLInputElement;

  @Element() el!: HTMLElement;

  /**
   * The type of control element
   */
  @Prop() role!: 'checkbox' | 'radio';

  /**
   * The primary text of this list item (required).
   */
  @Prop() text!: string;

  /**
   * Sets the secondary text of this list item.
   *
   * Requires `two-lines` on the parent `ino-list` element.
   */
  @Prop() secondaryText?: string;

  /**
   * Styles the row in a selected style.
   *
   * In contrast to `activated`, use this option to select one
   * or multiple items that are likely to change soon.
   */
  @Prop() selected?: boolean;

  /**
   * Styles the row in an activated style.
   *
   * In contrast to `selected`, use this for only one item
   * and to mark it as permanently activated.
   */
  @Prop() activated?: boolean;

  /**
   * Marks this element as checked. (**unmanaged**)
   */
  @Prop() checked?: boolean = false;

  /**
   * Disables this element.
   */
  @Prop() disabled?: boolean;

  /**
   * The name of this element.
   */
  @Prop() name?: string;

  /**
   * The value of this element.
   */
  @Prop() value?: string;

  /**
   * Marks this element as indeterminate (**unmanaged**)
   * Checkbox only
   */
  @Prop() indeterminate?: boolean;

  /**
   * Places the checkbox at the end of the item
   */
  @Prop() trailing?: boolean;

  /**
   * Emits when the user clicks on the checkbox or the list item to change the checked state. Contains the status in `event.detail`.
   */
  @Event() checkedChange!: EventEmitter;

  changedHandler = (e: CustomEvent<boolean>) => {
    e.stopPropagation();
    this.checkedChange.emit(e.detail);
  };

  clickHandler = (e: MouseEvent) => {
    e.stopPropagation();
    if (this.disabled || e.target['tagName'] === 'INO-CHECKBOX') {
      return;
    }

    this.inputEl.shadowRoot
      .querySelector('input')
      .dispatchEvent(new CustomEvent('input'));
  };

  render() {
    const controlItemPosition = this.trailing ? 'trailing' : 'leading';
    const slotPosition =
      this.el.children.length > 0
        ? this.trailing
          ? 'leading'
          : 'trailing'
        : '';

    const controlItemProps = {
      slot: controlItemPosition,
      checked: this.checked,
      disabled: this.disabled,
      name: this.name,
      value: this.value,
      onCheckedChange: this.changedHandler,
      ref: (inputEl) => (this.inputEl = inputEl),
    };

    return (
      <Host>
        <ino-list-item
          text={this.text}
          secondaryText={this.secondaryText}
          activated={this.activated}
          selected={this.selected}
          disabled={this.disabled}
          onClick={this.clickHandler}
        >
          {this.role === 'checkbox' ? (
            <ino-checkbox
              {...controlItemProps}
              indeterminate={this.indeterminate}
            />
          ) : (
            <ino-radio {...controlItemProps} />
          )}
          <span slot={slotPosition}>
            <slot></slot>
          </span>
        </ino-list-item>
      </Host>
    );
  }
}
