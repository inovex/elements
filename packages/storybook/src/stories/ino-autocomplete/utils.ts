import { KeyValue } from '@inovex.de/elements';

export const AutoCompleteStoryOptions: KeyValue[] = [
  {
    key: 'hh',
    value: 'Hamburg',
  },
  {
    key: 'ber',
    value: 'Berlin',
  },
  {
    key: 'k',
    value: 'Karlsruhe',
  },
];
