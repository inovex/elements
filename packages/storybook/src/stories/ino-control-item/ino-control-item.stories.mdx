import { Meta, Description } from "@storybook/addon-docs";
import readme from "../../../../elements/src/components/ino-control-item/readme.md";

<Meta title="Structure/ino-control-item/Documentation" />

<Description markdown={readme} />
