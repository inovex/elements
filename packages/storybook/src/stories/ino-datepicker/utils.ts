import '@inovex.de/elements';

export function registerInlineDatepickerHandler() {
  const group = document.querySelector('ino-radio-group') as HTMLInoRadioGroupElement;
  const datepicker = document.querySelector('.datepicker-group ino-datepicker') as HTMLInoDatepickerElement;

  const today = document.querySelector('.today');
  const thisWeek = document.querySelector('.thisWeek');
  const lastWeek = document.querySelector('.lastWeek');

  const todayClickHandler = () => {
    datepicker.value = '02.03.2020';
    datepicker.label = 'Am';
    group.value = 'at';
    datepicker.range = false;
  };

  const thisWeekClickHandler = () => {
    group.value = 'range';
    datepicker.label = 'Zeitraum';
    datepicker.range = true;
    datepicker.value = '02.03.2020 bis 08.03.2020';
  };

  const lastWeekClickHandler = () => {
    group.value = 'range';
    datepicker.label = 'Zeitraum';
    datepicker.range = true;
    datepicker.value = '24.02.2020 bis 01.03.2020';
  };

  const checkedChangeHandler = (e: CustomEvent<boolean>) => {
    const target = e.target as HTMLInoRadioElement;

    switch (target.value) {
      case 'at':
        datepicker.label = 'Am';
        datepicker.range = false;
        break;
      case 'after':
        datepicker.label = 'Ab';
        datepicker.range = false;
        if (group.value == 'range') {
          datepicker.value = datepicker.value.split(' - ').shift();
        }
        break;
      case 'before':
        datepicker.label = 'Bis';
        datepicker.range = false;
        if (group.value == 'range') {
          datepicker.value = datepicker.value.split(' - ').pop();
        }
        break;
      case 'range':
        datepicker.range = true;
        datepicker.label = 'Zeitraum';
        break;
    }
    group.value = target.value;
  };

  today.addEventListener('click', todayClickHandler);
  thisWeek.addEventListener('click', thisWeekClickHandler);
  lastWeek.addEventListener('click', lastWeekClickHandler);
  group.addEventListener('checkedChange', checkedChangeHandler);

  return () => {
    today.removeEventListener('click', todayClickHandler);
    thisWeek.removeEventListener('click', thisWeekClickHandler);
    lastWeek.removeEventListener('click', lastWeekClickHandler);
    group.removeEventListener('checkedChange', checkedChangeHandler);
  };
}
