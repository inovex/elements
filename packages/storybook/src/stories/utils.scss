/**
 * Applies styling to the container of a story.
 * The styling is encapsulated within this mixin because the ID may change in the future.
 */
@mixin story-container {
  #root-inner {
    @content;
  }
}
