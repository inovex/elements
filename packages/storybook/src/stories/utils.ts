import { html } from 'lit-html';
import ICONS from '../../../elements/src/components/ino-icon/icons';

export const maybeCreateStoryArgs = (story: any) => {
  if (!story.args) {
    story.args = {};
  }
  if (!story.argTypes) {
    story.argTypes = {};
  }
};

export const withIconControl = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    control: {
      type: 'select',
    },
    options: getIcons(),
  };
};

export const withColorScheme = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    control: {
      type: 'select',
    },
    options: [
      'primary',
      'secondary',
      'success',
      'warning',
      'error',
      'light',
      'dark',
    ],
  };
};

export const withSortDirection = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    options: ['desc', 'asc'],
    control: {
      type: 'select',
    },
  };
};

export const getIcons = () => ICONS.filter((icon) => !icon.startsWith('_'));

export const defaultDecorator = (story, className?: string) => {
  return html` <div class="${className ?? ''}">${story()}</div> `;
};
