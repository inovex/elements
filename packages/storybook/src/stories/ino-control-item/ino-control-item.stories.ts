import { Components } from '@inovex.de/elements';
import { useEffect } from '@storybook/client-api';
import { Story } from '@storybook/web-components';
import { html } from 'lit-html';
import { defaultDecorator } from '../utils';

import './ino-control-item.scss';

export default {
  title: 'Structure/ino-control-item',
  component: 'ino-control-item',
  decorators: [
    defaultDecorator,
    (story) => {
      useEffect(() => {
        const checkedChangeHandler = (e: CustomEvent<boolean>) => {
          const controlItem = e.target as HTMLInoControlItemElement;

          controlItem.checked = e.detail;

          if (controlItem.indeterminate) {
            controlItem.indeterminate = false;
          }
        };
        document.addEventListener('checkedChange', checkedChangeHandler);
        return () =>
          document.removeEventListener('checkedChange', checkedChangeHandler);
      });
      return story();
    },
  ],
};

export const Playground: Story<Components.InoControlItem> = (args) => html`
  <ino-list>
    <ino-control-item
      activated="${args.activated}"
      checked="${args.checked}"
      disabled="${args.disabled}"
      indeterminate="${args.indeterminate}"
      role="${args.role}"
      secondaryText="${args.secondaryText}"
      selected="${args.selected}"
      text="${args.text}"
      trailing="${args.trailing}"
    >
    </ino-control-item>
  </ino-list>
`;

Playground.args = {
  activated: false,
  checked: false,
  disabled: false,
  indeterminate: false,
  role: 'checkbox',
  secondaryText: 'Secondary Text',
  selected: false,
  text: 'Text',
  trailing: false,
};

Playground.argTypes = {
  role: {
    control: {
      type: 'select',
    },
    options: ['checkbox', 'radio'],
  },
};

export const Roles = () => html`
  <ino-list>
    <ino-control-item checked role="checkbox" text="Checkbox">
    </ino-control-item>
  </ino-list>
  <ino-list>
    <ino-control-item role="radio" text="Radio"></ino-control-item>
  </ino-list>
`;

export const States = () => html`
  <ino-list>
    <ino-control-item role="checkbox" text="Checked" checked>
    </ino-control-item>
    <ino-control-item role="checkbox" text="Indeterminate" indeterminate>
    </ino-control-item>
    <ino-control-item role="checkbox" text="Activated" activated>
    </ino-control-item>
    <ino-control-item role="checkbox" text="Selected" selected>
    </ino-control-item>
    <ino-control-item role="checkbox" text="Disabled" disabled>
    </ino-control-item>
  </ino-list>
`;

export const Position = () => html`
  <ino-list>
    <ino-control-item role="checkbox" text="Leading"> </ino-control-item>
    <ino-control-item trailing role="checkbox" text="Trailing">
    </ino-control-item>
  </ino-list>
`;
