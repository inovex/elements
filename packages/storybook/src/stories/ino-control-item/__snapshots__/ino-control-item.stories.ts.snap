// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Structure/ino-control-item Activated smoke-test 1`] = `
<div id="root-inner">
  <div class>
    <ino-list two-lines="false"
              class="hydrated"
    >
      <ul class="mdc-deprecated-list">
        <ino-control-item activated="true"
                          checked="false"
                          disabled="false"
                          indeterminate="false"
                          role="checkbox"
                          secondary-text
                          selected="false"
                          text="Text"
                          trailing="false"
                          class="hydrated"
        >
          <ino-list-item class="hydrated">
            <span slot
                  hidden
            >
            </span>
            <li class="mdc-deprecated-list-item mdc-deprecated-list-item--activated mdc-ripple-upgraded"
                tabindex="-1"
            >
              <span class="mdc-deprecated-list-item__ripple">
              </span>
              <span class="mdc-deprecated-list-item__graphic"
                    role="presentation"
              >
                <ino-checkbox slot="leading"
                              class="hydrated"
                >
                  <input type="hidden"
                         class="aux-input"
                         name="undefined"
                         value
                  >
                </ino-checkbox>
              </span>
              <span class="mdc-deprecated-list-item__text">
                Text
              </span>
            </li>
          </ino-list-item>
        </ino-control-item>
      </ul>
    </ino-list>
  </div>
</div>
`;

exports[`Structure/ino-control-item Checked smoke-test 1`] = `
<div id="root-inner">
  <div class>
    <ino-list two-lines="false"
              class="hydrated"
    >
      <ul class="mdc-deprecated-list">
        <ino-control-item activated="false"
                          checked="true"
                          disabled="false"
                          indeterminate="false"
                          role="checkbox"
                          secondary-text
                          selected="false"
                          text="Text"
                          trailing="false"
                          class="hydrated"
        >
          <ino-list-item class="hydrated">
            <span slot
                  hidden
            >
            </span>
            <li class="mdc-deprecated-list-item mdc-ripple-upgraded"
                tabindex="-1"
            >
              <span class="mdc-deprecated-list-item__ripple">
              </span>
              <span class="mdc-deprecated-list-item__graphic"
                    role="presentation"
              >
                <ino-checkbox slot="leading"
                              class="hydrated"
                >
                  <input type="hidden"
                         class="aux-input"
                         name="undefined"
                         value
                  >
                </ino-checkbox>
              </span>
              <span class="mdc-deprecated-list-item__text">
                Text
              </span>
            </li>
          </ino-list-item>
        </ino-control-item>
      </ul>
    </ino-list>
  </div>
</div>
`;

exports[`Structure/ino-control-item Disabled smoke-test 1`] = `
<div id="root-inner">
  <div class>
    <ino-list two-lines="false">
      <ino-control-item activated="false"
                        checked="false"
                        disabled="true"
                        indeterminate="false"
                        role="checkbox"
                        secondary-text
                        selected="false"
                        text="Text"
                        trailing="false"
      >
      </ino-control-item>
    </ino-list>
  </div>
</div>
`;

exports[`Structure/ino-control-item Indeterminate smoke-test 1`] = `
<div id="root-inner">
  <div class>
    <ino-list two-lines="false"
              class="hydrated"
    >
      <ul class="mdc-deprecated-list">
        <ino-control-item activated="false"
                          checked="false"
                          disabled="false"
                          indeterminate="true"
                          role="checkbox"
                          secondary-text
                          selected="false"
                          text="Text"
                          trailing="false"
                          class="hydrated"
        >
          <ino-list-item class="hydrated">
            <span slot
                  hidden
            >
            </span>
            <li class="mdc-deprecated-list-item mdc-ripple-upgraded"
                tabindex="-1"
            >
              <span class="mdc-deprecated-list-item__ripple">
              </span>
              <span class="mdc-deprecated-list-item__graphic"
                    role="presentation"
              >
                <ino-checkbox slot="leading"
                              class="hydrated"
                >
                  <input type="hidden"
                         class="aux-input"
                         name="undefined"
                         value
                  >
                </ino-checkbox>
              </span>
              <span class="mdc-deprecated-list-item__text">
                Text
              </span>
            </li>
          </ino-list-item>
        </ino-control-item>
      </ul>
    </ino-list>
  </div>
</div>
`;

exports[`Structure/ino-control-item Playground smoke-test 1`] = `
<div id="root-inner">
  <div class>
    <ino-list two-lines="false"
              class="hydrated"
    >
      <ul class="mdc-deprecated-list">
        <ino-control-item activated="false"
                          checked="false"
                          disabled="false"
                          indeterminate="false"
                          role="checkbox"
                          secondary-text
                          selected="false"
                          text="Text"
                          trailing="false"
                          class="hydrated"
        >
          <ino-list-item class="hydrated">
            <span slot
                  hidden
            >
            </span>
            <li class="mdc-deprecated-list-item mdc-ripple-upgraded"
                tabindex="-1"
            >
              <span class="mdc-deprecated-list-item__ripple">
              </span>
              <span class="mdc-deprecated-list-item__graphic"
                    role="presentation"
              >
                <ino-checkbox slot="leading"
                              class="hydrated"
                >
                  <input type="hidden"
                         class="aux-input"
                         name="undefined"
                         value
                  >
                </ino-checkbox>
              </span>
              <span class="mdc-deprecated-list-item__text">
                Text
              </span>
            </li>
          </ino-list-item>
        </ino-control-item>
      </ul>
    </ino-list>
  </div>
</div>
`;

exports[`Structure/ino-control-item Roles smoke-test 1`] = `
<div id="root-inner">
  <div class>
    <ino-list two-lines="false"
              class="hydrated"
    >
      <ul class="mdc-deprecated-list">
        <ino-control-item activated="false"
                          checked="false"
                          disabled="false"
                          indeterminate="false"
                          role="radio"
                          secondary-text
                          selected="false"
                          text="Text"
                          trailing="false"
                          class="hydrated"
        >
          <ino-list-item class="hydrated">
            <span slot
                  hidden
            >
            </span>
            <li class="mdc-deprecated-list-item mdc-ripple-upgraded"
                tabindex="-1"
            >
              <span class="mdc-deprecated-list-item__ripple">
              </span>
              <span class="mdc-deprecated-list-item__graphic"
                    role="presentation"
              >
                <ino-radio slot="leading"
                           class="hydrated"
                >
                  <input type="hidden"
                         class="aux-input"
                         name="undefined"
                         value
                  >
                </ino-radio>
              </span>
              <span class="mdc-deprecated-list-item__text">
                Text
              </span>
            </li>
          </ino-list-item>
        </ino-control-item>
      </ul>
    </ino-list>
  </div>
</div>
`;

exports[`Structure/ino-control-item SecondaryText smoke-test 1`] = `
<div id="root-inner">
  <div class>
    <ino-list two-lines="Some Secondary Text"
              class="hydrated"
    >
      <ul class="mdc-deprecated-list mdc-deprecated-list--two-line">
        <ino-control-item activated="false"
                          checked="false"
                          disabled="false"
                          indeterminate="false"
                          role="checkbox"
                          secondary-text="Some Secondary Text"
                          selected="false"
                          text="Text"
                          trailing="false"
                          class="hydrated"
        >
          <ino-list-item class="hydrated">
            <span slot
                  hidden
            >
            </span>
            <li class="mdc-deprecated-list-item"
                tabindex="-1"
            >
              <span class="mdc-deprecated-list-item__ripple">
              </span>
              <span class="mdc-deprecated-list-item__graphic"
                    role="presentation"
              >
                <ino-checkbox slot="leading"
                              class="hydrated"
                >
                  <input type="hidden"
                         class="aux-input"
                         name="undefined"
                         value
                  >
                </ino-checkbox>
              </span>
              <span class="mdc-deprecated-list-item__text">
                <span class="mdc-deprecated-list-item__primary-text">
                  Text
                </span>
                <span class="mdc-deprecated-list-item__secondary-text">
                  Some Secondary Text
                </span>
              </span>
            </li>
          </ino-list-item>
        </ino-control-item>
      </ul>
    </ino-list>
  </div>
</div>
`;

exports[`Structure/ino-control-item Selected smoke-test 1`] = `
<div id="root-inner">
  <div class>
    <ino-list two-lines="false"
              class="hydrated"
    >
      <ul class="mdc-deprecated-list">
        <ino-control-item activated="false"
                          checked="false"
                          disabled="false"
                          indeterminate="false"
                          role="checkbox"
                          secondary-text
                          selected="true"
                          text="Text"
                          trailing="false"
                          class="hydrated"
        >
          <ino-list-item class="hydrated">
            <span slot
                  hidden
            >
            </span>
            <li class="mdc-deprecated-list-item mdc-deprecated-list-item--selected"
                tabindex="-1"
            >
              <span class="mdc-deprecated-list-item__ripple">
              </span>
              <span class="mdc-deprecated-list-item__graphic"
                    role="presentation"
              >
                <ino-checkbox slot="leading"
                              class="hydrated"
                >
                  <input type="hidden"
                         class="aux-input"
                         name="undefined"
                         value
                  >
                </ino-checkbox>
              </span>
              <span class="mdc-deprecated-list-item__text">
                Text
              </span>
            </li>
          </ino-list-item>
        </ino-control-item>
      </ul>
    </ino-list>
  </div>
</div>
`;

exports[`Structure/ino-control-item Trailing smoke-test 1`] = `
<div id="root-inner">
  <div class>
    <ino-list two-lines="false">
      <ino-control-item activated="false"
                        checked="false"
                        disabled="false"
                        indeterminate="false"
                        role="checkbox"
                        secondary-text
                        selected="false"
                        text="Text"
                        trailing="true"
      >
      </ino-control-item>
    </ino-list>
  </div>
</div>
`;
